src/Lib/Models/CanvasItem.vala

```diff
@@ -106,9 +123,103 @@ public interface Akira.Lib.Models.CanvasItem : Goo.CanvasItemSimple, Goo.CanvasI
+        //var z_index = (item as Models.CanvasItej).canvas.get_root_item ().find_child (item);
+
+        //item.set ("z-index", z_index);
+    }
+
+    public virtual void position_item (double _x, double _y) {
+      if (artboard != null) {
+        artboard.add_child (this, -1);
+
+        double item_x_from_artboard = _x;
+        double item_y_from_artboard = _y;
+
+        canvas.convert_to_item_space (artboard, ref item_x_from_artboard, ref item_y_from_artboard);
+
+        relative_x = item_x_from_artboard;
+        relative_y = item_y_from_artboard;
+      } else {
+        parent.add_child (this, -1);
+
+
+        // Keep the item always in the origin
+        // move the entire coordinate system every time
+        translate (_x, _y);
+      }
+    }
+
+
+    public virtual void move (
+      double delta_x, double delta_y,
+      double delta_x_accumulator = 0.0, double delta_y_accumulator = 0.0) {
+      if (this is Models.CanvasArtboard) {
+        (this as Models.CanvasArtboard).move_items (delta_x, delta_y);
+      }
+
+      if (this.artboard != null) {
+        var transformed_delta_x = delta_x_accumulator;
+        var transformed_delta_y = delta_y_accumulator;
+
+        this.relative_x = this.initial_relative_x + transformed_delta_x;
+        this.relative_y = this.initial_relative_y + transformed_delta_y;
+
+
+        return;
+      }
 
-        item.set ("z-index", z_index);
+      this.translate (delta_x, delta_y);
+    }
+
+    public virtual Cairo.Matrix get_real_transform () {
+      Cairo.Matrix transform = Cairo.Matrix.identity ();
+
+      if (artboard == null) {
+        get_transform (out transform);
+      } else {
+        artboard.get_transform (out transform);
+
+        transform = compute_transform (transform);
+      }
+
+      return transform;
+    }
+
+    public virtual Cairo.Matrix compute_transform (Cairo.Matrix transform) {
+        transform.translate (relative_x, relative_y);
+
+        var width = get_coords ("width");
+        var height = get_coords ("height");
+
+        // Rotate around the center by the amount
+        // in item.rotation
+        transform.translate (width / 2, height / 2);
+        transform.rotate (Utils.AffineTransform.deg_to_rad (rotation));
+        transform.translate (- (width / 2), - (height / 2));
+
+        return transform;
+    }
+
+    public virtual double get_real_coord (string coord_id) {
+      var offset_x = this.artboard == null ? 0 : relative_x;
+      var offset_y = this.artboard == null ? 0 : relative_y;
+
+      var item_x = get_coords ("x") - offset_x;
+      var item_y = get_coords ("y") - offset_y;
+
+      switch (coord_id) {
+        case "x":
+          return item_x;
+        case "y":
+          return item_y;
+        default:
+          return 0.0;
+      }
+    }
+
+    public virtual void store_relative_position () {
+      this.initial_relative_x = this.relative_x;
+      this.initial_relative_y = this.relative_y;
```

Comment:
This whole section has wrong indentation.

Reply:
Yes, I had the same feeling. Something has gone wrong with my editor! I'll try to revert to the `editorconfig` settings.